docs/reference.md

```diff
@@ -2122,14 +2123,75 @@ protected:
 See "[Writing custom tracers](CookBook.md/#custom_tracer)" in the
 [Cook Book](CookBook.md) for an example.
 
-### <A name="typed_matcher"/>`trompeloeil::typed_matcher<T>`
+### <A name="typed_matcher"/> `trompeloeil::typed_matcher<T>`
 
 Convenience class available when writing custom matchers for a specific
 type. It inherits from [`trompeloeil::matcher`](#matcher_type).
 
 See "[Writing custom matchers](CookBook.md/#custom_matchers)" in the
 [Cook Book](CookBook.md) for examples.
 
+### <A named="movable_mock"/> `trompeloeil_movable_mock`
+
+By adding a public member type `trompeloeil_movable_mock` to your mock
+struct/class, you make it move constructible. Note that when a mock object is
+moved, any current expectations will be taken over by the newly constructed
+mock object, but the implicitly created lambdas associated with
+[**`.WITH()`**](reference.md/#WITH),
+[**`.SIDE_EFFECT()`**](reference.md/#SIDE_EFFECT),
+[**`.RETURN()`**](reference.md/#RETURN) and
+[**`.THROW()`**](reference.md/#THROW) and their `**LR_**` counter parts, will
+not themselves be moved. This means that if they refer to member variables
+in the mock objects, they will refer to dead store after the move. Also, keep
+in mind the lifetime of expectations. If you move a mock object that has
+active expectations, they should almost certainly be
+[**`NAMED_REQUIRE_CALL()`**](reference.md/#NAMED_REQUIRE_CALL),
+[**`NAMED_ALLOW_CALL()`**](reference.md/#NAMED_ALLOW_CALL) or
+[**`NAMED_FORBID_CALL()`**](reference.md/#NAMED_FORBID_CALL).
+
+```Cpp
+class immobile
+{
+public:
+  MAKE_MOCK1(func, void(int));
+};
+
+class movable
+{
+public:
+  int i = 0;
+  struct trompeloeil_movable_mock{}; // allow move construction
+  MAKE_MOCK1(func, void(int));
+};
+
+template <typename T>
+T transfer(T t)
+{
+  return t;
+}
+
+test(...)
+{
+  auto m = transfer(immobile{}); // compilation error
+  ...
+}
+test(...)
+{
+  movable m;
+  auto e = NAMED_REQUIRE_CALL(m, func(3));
+  auto mm = transfer(std::move(m)); // OK, e now belongs to mm.
```

Comment:
I'm confused by this - is there more than one kind of ownership involved here? `e` is a `unique_ptr`, so _it_ is owning something, in some sense. Does the mock object (`m` or `mm`) also own the expectation?

Reply:
Wrong word there. `e now refers to mm` would probably be better, or `A call to mm->func(3) will now satisfy e`, or something like that.